core/src/main/java/jenkins/util/io/PathRemover.java

```diff
@@ -285,6 +289,15 @@ private static void makeWritable(@Nonnull Path path) throws IOException {
             } catch (UnsupportedOperationException ignored) {
                 // PosixFileAttributes not supported, fall back to old IO.
             }
+        } else {
+          /*
+           * If on Windows a folder has a read only attribute set, the file.setWritable(true) doesn't work (JENKINS-57855)
+           */
+          DosFileAttributeView dos = Files.getFileAttributeView(path, DosFileAttributeView.class, LinkOption.NOFOLLOW_LINKS);
+          if (dos != null)
```

Comment:
Is it supposed to fall through and call `path.toFile().setWritable(true);` in such case? Looks so

Reply:
I would assume that it is not necessary to call the setWritable(true) for windows.
If this fails then the setWritable would also fail I guess.